`test_add_layernorm_add_gemm_nonstd` fails to compile when LayerNorm fusions are disabled

Reproduces on the develop branch with 
` MIGRAPHX_DISABLE_LAYERNORM_FUSION=1 MIGRAPHX_TRACE_TEST_COMPILE="gpu"   ./bin/test_verify "test_add_layernorm_add_gemm_nonstd" `

```
In file included from main.cpp:3:
./migraphx/kernels/reduce.hpp:674:21: error: assigning to 'type' (aka 'float') from incompatible type 'array<decltype(float{} + float{}), 2U>' (aka 'array<float, 2U>')
                r = op(r, read(x(j, _c<0>), xs(j, _c<0>)...));
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./migraphx/kernels/reduce.hpp:420:28: note: in instantiation of function template specialization 'migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>::reduce_impl<migraphx::op::sum, int, (lambda at ./migraphx/kernels/functional.hpp:217:16), migraphx::integral_constant<unsigned int, 16>, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:411:47)> &, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:411:47)> &>' requested here
            return derived.reduce_impl(op, init, read, n, xs...);
                           ^
./migraphx/kernels/reduce.hpp:384:43: note: in instantiation of function template specialization 'migraphx::reduce::reducer_base<migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>>::reduce(migraphx::op::sum, int, (lambda at ./migraphx/kernels/functional.hpp:217:16))::(anonymous class)::operator()<migraphx::integral_constant<unsigned int, 16>, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:411:47)> &, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:411:47)> &>' requested here
        return [=](auto&&... xs) { return f(get_size(xs...), make_inner_slice(xs)...); };
                                          ^
main.cpp:100:21: note: in instantiation of function template specialization 'migraphx::reduce::reducer_base<migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>>::inner_sliced((lambda at ./migraphx/kernels/reduce.hpp:418:35))::(anonymous class)::operator()<migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:411:47)> &, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:411:47)> &>' requested here
auto zz7 = op::id{}(r.reduce(op::sum{}, 0, compose(array_apply(op::id{}), MIGRAPHX_LIFT(make_array)))(zz6, zz5));
                    ^
main.cpp:124:138: note: in instantiation of function template specialization 'migraphx::fused_reduce_op<migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>, migraphx::array<unsigned int, 2>>' requested here
        fused_reduce<reduce::lane, decltype(make_shape(index_ints<1, 8>{}, index_ints<8, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
                                                                                                                                         ^
./migraphx/kernels/functional.hpp:225:43: note: while substituting deduced template arguments into function template 'operator()' [with private_lifts_xs:auto = <const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>> &, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)> &, migraphx::array<unsigned int, 2> &>]
        return [=](auto&&... ys) { return f(xs..., static_cast<decltype(ys)>(ys)...); };
                                          ^
./migraphx/kernels/reduce.hpp:760:23: note: in instantiation of function template specialization 'migraphx::partial((lambda at main.cpp:124:124))::(anonymous class)::operator()(migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>)::(anonymous class)::operator()<migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)> &, migraphx::array<unsigned int, 2> &>' requested here
        auto result = f(r, out_idx);
                      ^
./migraphx/kernels/reduce.hpp:713:13: note: in instantiation of function template specialization 'migraphx::fused_reduce(migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::assign_none, (lambda at ./migraphx/kernels/functional.hpp:225:16))::(anonymous class)::operator()<migraphx::array<unsigned int, 2>, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>>' requested here
            f(out_idx, make(idx, [&](auto input) { return reduce_slice<Output>(input, out_idx); }));
            ^
./migraphx/kernels/index.hpp:200:64: note: in instantiation of function template specialization 'migraphx::reduce::lane::run((lambda at ./migraphx/kernels/reduce.hpp:759:33))::(anonymous class)::operator()<unsigned int>' requested here
    static constexpr auto invoke_loop(F f, I i, D) -> decltype(f(i))
                                                               ^
./migraphx/kernels/index.hpp:244:21: note: while substituting deduced template arguments into function template 'invoke_loop' [with F = (lambda at ./migraphx/kernels/reduce.hpp:711:38), I = index_int, D = integral_constant<decltype(0), 0>]
                    invoke_loop(f, start, _c<0>);
                    ^
./migraphx/kernels/index.hpp:268:9: note: in instantiation of function template specialization 'migraphx::index::for_stride<false, (lambda at ./migraphx/kernels/reduce.hpp:711:38), migraphx::integral_constant<unsigned int, 8>, migraphx::integral_constant<unsigned int, 8>>' requested here
        for_stride<false>(global, n, nglobal(), f);
        ^
./migraphx/kernels/reduce.hpp:711:13: note: in instantiation of function template specialization 'migraphx::index::global_stride<(lambda at ./migraphx/kernels/reduce.hpp:711:38), migraphx::integral_constant<unsigned int, 8>>' requested here
        idx.global_stride(nelements, [&](auto i) {
            ^
./migraphx/kernels/reduce.hpp:759:20: note: in instantiation of function template specialization 'migraphx::reduce::lane::run<migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>, (lambda at ./migraphx/kernels/reduce.hpp:759:33)>' requested here
    Algo::template run<Reduced>([&](auto out_idx, auto r) {
                   ^
main.cpp:124:9: note: in instantiation of function template specialization 'migraphx::fused_reduce<migraphx::reduce::lane, migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::assign_none, (lambda at ./migraphx/kernels/functional.hpp:225:16)>' requested here
        fused_reduce<reduce::lane, decltype(make_shape(index_ints<1, 8>{}, index_ints<8, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
        ^
main.cpp:102:12: error: subscripted value is not an array, pointer, or vector
auto zz8 = zz7[1];
           ^~~ ~
main.cpp:124:138: note: in instantiation of function template specialization 'migraphx::fused_reduce_op<migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>, migraphx::array<unsigned int, 2>>' requested here
        fused_reduce<reduce::lane, decltype(make_shape(index_ints<1, 8>{}, index_ints<8, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
                                                                                                                                         ^
./migraphx/kernels/functional.hpp:225:43: note: while substituting deduced template arguments into function template 'operator()' [with private_lifts_xs:auto = <const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>> &, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)> &, migraphx::array<unsigned int, 2> &>]
        return [=](auto&&... ys) { return f(xs..., static_cast<decltype(ys)>(ys)...); };
                                          ^
./migraphx/kernels/reduce.hpp:760:23: note: in instantiation of function template specialization 'migraphx::partial((lambda at main.cpp:124:124))::(anonymous class)::operator()(migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>)::(anonymous class)::operator()<migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)> &, migraphx::array<unsigned int, 2> &>' requested here
        auto result = f(r, out_idx);
                      ^
./migraphx/kernels/reduce.hpp:713:13: note: in instantiation of function template specialization 'migraphx::fused_reduce(migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::assign_none, (lambda at ./migraphx/kernels/functional.hpp:225:16))::(anonymous class)::operator()<migraphx::array<unsigned int, 2>, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>>' requested here
            f(out_idx, make(idx, [&](auto input) { return reduce_slice<Output>(input, out_idx); }));
            ^
./migraphx/kernels/index.hpp:200:64: note: in instantiation of function template specialization 'migraphx::reduce::lane::run((lambda at ./migraphx/kernels/reduce.hpp:759:33))::(anonymous class)::operator()<unsigned int>' requested here
    static constexpr auto invoke_loop(F f, I i, D) -> decltype(f(i))
                                                               ^
./migraphx/kernels/index.hpp:244:21: note: while substituting deduced template arguments into function template 'invoke_loop' [with F = (lambda at ./migraphx/kernels/reduce.hpp:711:38), I = index_int, D = integral_constant<decltype(0), 0>]
                    invoke_loop(f, start, _c<0>);
                    ^
./migraphx/kernels/index.hpp:268:9: note: in instantiation of function template specialization 'migraphx::index::for_stride<false, (lambda at ./migraphx/kernels/reduce.hpp:711:38), migraphx::integral_constant<unsigned int, 8>, migraphx::integral_constant<unsigned int, 8>>' requested here
        for_stride<false>(global, n, nglobal(), f);
        ^
./migraphx/kernels/reduce.hpp:711:13: note: in instantiation of function template specialization 'migraphx::index::global_stride<(lambda at ./migraphx/kernels/reduce.hpp:711:38), migraphx::integral_constant<unsigned int, 8>>' requested here
        idx.global_stride(nelements, [&](auto i) {
            ^
./migraphx/kernels/reduce.hpp:759:20: note: in instantiation of function template specialization 'migraphx::reduce::lane::run<migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>, (lambda at ./migraphx/kernels/reduce.hpp:759:33)>' requested here
    Algo::template run<Reduced>([&](auto out_idx, auto r) {
                   ^
main.cpp:124:9: note: in instantiation of function template specialization 'migraphx::fused_reduce<migraphx::reduce::lane, migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::assign_none, (lambda at ./migraphx/kernels/functional.hpp:225:16)>' requested here
        fused_reduce<reduce::lane, decltype(make_shape(index_ints<1, 8>{}, index_ints<8, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
        ^
main.cpp:104:12: error: subscripted value is not an array, pointer, or vector
auto zz9 = zz7[0];
           ^~~ ~
In file included from main.cpp:3:
./migraphx/kernels/reduce.hpp:697:76: error: no matching function for call to object of type 'const (lambda at main.cpp:112:21)'
            return make_lazy_inner_storage(n, [=](auto j, auto d) { return f(xs(j, d)...); });
                                                                           ^
./migraphx/kernels/reduce.hpp:276:46: note: in instantiation of function template specialization 'migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>::inner_impl((lambda at main.cpp:112:21), migraphx::integral_constant<unsigned int, 16>, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)> &, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)> &, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)> &)::(anonymous class)::operator()<int, migraphx::integral_constant<int, 0>>' requested here
    using type = remove_reference_t<decltype(declval<F>()(0, _c<0>))>;
                                             ^
./migraphx/kernels/reduce.hpp:287:39: note: in instantiation of template class 'migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)>' requested here
constexpr lazy_inner_storage<Size, F> make_lazy_inner_storage(Size, F f)
                                      ^
./migraphx/kernels/reduce.hpp:697:20: note: in instantiation of function template specialization 'migraphx::reduce::make_lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)>' requested here
            return make_lazy_inner_storage(n, [=](auto j, auto d) { return f(xs(j, d)...); });
                   ^
./migraphx/kernels/reduce.hpp:402:41: note: in instantiation of function template specialization 'migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>::inner_impl<int, (lambda at main.cpp:112:21), migraphx::integral_constant<unsigned int, 16>, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)> &, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)> &, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)> &>' requested here
                return derived.template inner_impl<result_type>(f, n, xs...);
                                        ^
./migraphx/kernels/reduce.hpp:384:43: note: in instantiation of function template specialization 'migraphx::reduce::reducer_base<migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>>::inner((lambda at main.cpp:112:21))::(anonymous class)::operator()<migraphx::integral_constant<unsigned int, 16>, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)> &, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)>, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)>>' requested here
        return [=](auto&&... xs) { return f(get_size(xs...), make_inner_slice(xs)...); };
                                          ^
main.cpp:112:13: note: in instantiation of function template specialization 'migraphx::reduce::reducer_base<migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>>::inner_sliced((lambda at ./migraphx/kernels/reduce.hpp:393:35))::(anonymous class)::operator()<migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)> &, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>> &, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>> &>' requested here
auto zz13 = r.inner([=](auto zz4_lambda_param, auto x2_lambda_param, auto x3_lambda_param) { return pointwise4(zz4_lambda_param, zz12, zz11, x2_lambda_param, x3_lambda_param); })(zz4, x2, x3);
            ^
main.cpp:124:138: note: in instantiation of function template specialization 'migraphx::fused_reduce_op<migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>, migraphx::array<unsigned int, 2>>' requested here
        fused_reduce<reduce::lane, decltype(make_shape(index_ints<1, 8>{}, index_ints<8, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
                                                                                                                                         ^
./migraphx/kernels/functional.hpp:225:43: note: while substituting deduced template arguments into function template 'operator()' [with private_lifts_xs:auto = <const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>> &, const migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>> &, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)> &, migraphx::array<unsigned int, 2> &>]
        return [=](auto&&... ys) { return f(xs..., static_cast<decltype(ys)>(ys)...); };
                                          ^
./migraphx/kernels/reduce.hpp:760:23: note: in instantiation of function template specialization 'migraphx::partial((lambda at main.cpp:124:124))::(anonymous class)::operator()(migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 1, 0>>>)::(anonymous class)::operator()<migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)> &, migraphx::array<unsigned int, 2> &>' requested here
        auto result = f(r, out_idx);
                      ^
./migraphx/kernels/reduce.hpp:713:13: note: in instantiation of function template specialization 'migraphx::fused_reduce(migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::assign_none, (lambda at ./migraphx/kernels/functional.hpp:225:16))::(anonymous class)::operator()<migraphx::array<unsigned int, 2>, migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>>' requested here
            f(out_idx, make(idx, [&](auto input) { return reduce_slice<Output>(input, out_idx); }));
            ^
./migraphx/kernels/index.hpp:200:64: note: in instantiation of function template specialization 'migraphx::reduce::lane::run((lambda at ./migraphx/kernels/reduce.hpp:759:33))::(anonymous class)::operator()<unsigned int>' requested here
    static constexpr auto invoke_loop(F f, I i, D) -> decltype(f(i))
                                                               ^
./migraphx/kernels/index.hpp:244:21: note: while substituting deduced template arguments into function template 'invoke_loop' [with F = (lambda at ./migraphx/kernels/reduce.hpp:711:38), I = index_int, D = integral_constant<decltype(0), 0>]
                    invoke_loop(f, start, _c<0>);
                    ^
./migraphx/kernels/index.hpp:268:9: note: in instantiation of function template specialization 'migraphx::index::for_stride<false, (lambda at ./migraphx/kernels/reduce.hpp:711:38), migraphx::integral_constant<unsigned int, 8>, migraphx::integral_constant<unsigned int, 8>>' requested here
        for_stride<false>(global, n, nglobal(), f);
        ^
./migraphx/kernels/reduce.hpp:711:13: note: in instantiation of function template specialization 'migraphx::index::global_stride<(lambda at ./migraphx/kernels/reduce.hpp:711:38), migraphx::integral_constant<unsigned int, 8>>' requested here
        idx.global_stride(nelements, [&](auto i) {
            ^
./migraphx/kernels/reduce.hpp:759:20: note: in instantiation of function template specialization 'migraphx::reduce::lane::run<migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>, (lambda at ./migraphx/kernels/reduce.hpp:759:33)>' requested here
    Algo::template run<Reduced>([&](auto out_idx, auto r) {
                   ^
main.cpp:124:9: note: in instantiation of function template specialization 'migraphx::fused_reduce<migraphx::reduce::lane, migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::assign_none, (lambda at ./migraphx/kernels/functional.hpp:225:16)>' requested here
        fused_reduce<reduce::lane, decltype(make_shape(index_ints<1, 8>{}, index_ints<8, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
        ^
main.cpp:112:21: note: candidate template ignored: substitution failure [with zz4_lambda_param:auto = float, x2_lambda_param:auto = float, x3_lambda_param:auto = float]
auto zz13 = r.inner([=](auto zz4_lambda_param, auto x2_lambda_param, auto x3_lambda_param) { return pointwise4(zz4_lambda_param, zz12, zz11, x2_lambda_param, x3_lambda_param); })(zz4, x2, x3);
                    ^
In file included from main.cpp:3:
./migraphx/kernels/reduce.hpp:697:76: error: no matching function for call to object of type 'const (lambda at main.cpp:112:21)'
            return make_lazy_inner_storage(n, [=](auto j, auto d) { return f(xs(j, d)...); });
                                                                           ^
./migraphx/kernels/reduce.hpp:282:16: note: in instantiation of function template specialization 'migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>::inner_impl((lambda at main.cpp:112:21), migraphx::integral_constant<unsigned int, 16>, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)> &, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)> &, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)> &)::(anonymous class)::operator()<unsigned int, migraphx::integral_constant<int, 0>>' requested here
        return f(j, d);
               ^
./migraphx/kernels/reduce.hpp:690:19: note: in instantiation of function template specialization 'migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)>::operator()<unsigned int, migraphx::integral_constant<int, 0>>' requested here
                f(xs(j, _c<0>)...);
                  ^
./migraphx/kernels/reduce.hpp:398:25: note: in instantiation of function template specialization 'migraphx::reduce::lane::reducer<(lambda at ./migraphx/kernels/reduce.hpp:713:34)>::inner_void_impl<(lambda at ./migraphx/kernels/reduce.hpp:763:21), migraphx::integral_constant<unsigned int, 16>, migraphx::reduce::storage_access<float, (lambda at ./migraphx/kernels/reduce.hpp:355:17)> &, migraphx::reduce::lazy_inner_storage<migraphx::integral_constant<unsigned int, 16>, (lambda at ./migraphx/kernels/reduce.hpp:697:47)> &>' requested here
                derived.inner_void_impl(f, n, xs...);
                        ^
./migraphx/kernels/reduce.hpp:711:13: note: in instantiation of function template specialization 'migraphx::index::global_stride<(lambda at ./migraphx/kernels/reduce.hpp:711:38), migraphx::integral_constant<unsigned int, 8>>' requested here
        idx.global_stride(nelements, [&](auto i) {
            ^
./migraphx/kernels/reduce.hpp:759:20: note: in instantiation of function template specialization 'migraphx::reduce::lane::run<migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>, (lambda at ./migraphx/kernels/reduce.hpp:759:33)>' requested here
    Algo::template run<Reduced>([&](auto out_idx, auto r) {
                   ^
main.cpp:124:9: note: in instantiation of function template specialization 'migraphx::fused_reduce<migraphx::reduce::lane, migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>, migraphx::tensor_view<float, migraphx::shape<migraphx::integral_const_array<unsigned int, 16, 8>, migraphx::integral_const_array<unsigned int, 8, 1>>>, migraphx::assign_none, (lambda at ./migraphx/kernels/functional.hpp:225:16)>' requested here
        fused_reduce<reduce::lane, decltype(make_shape(index_ints<1, 8>{}, index_ints<8, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
        ^
main.cpp:112:21: note: candidate template ignored: substitution failure [with zz4_lambda_param:auto = float, x2_lambda_param:auto = float, x3_lambda_param:auto = float]
auto zz13 = r.inner([=](auto zz4_lambda_param, auto x2_lambda_param, auto x3_lambda_param) { return pointwise4(zz4_lambda_param, zz12, zz11, x2_lambda_param, x3_lambda_param); })(zz4, x2, x3);
                    ^
5 errors generated when compiling for gfx908.

```

--- a/src/targets/gpu/kernels/include/migraphx/kernels/reduce.hpp
+++ b/src/targets/gpu/kernels/include/migraphx/kernels/reduce.hpp
@@ -667,7 +667,7 @@ struct lane
         template <class Op, class T, class Read, class N, class U, class... Us>
         __device__ auto reduce_impl(Op op, T init, Read read, N n, U&& x, Us&&... xs) const
         {
-            using type = remove_reference_t<decltype(x(0, _c<0>))>;
+            using type = remove_reference_t<decltype(read(x(0, _c<0>), xs(0, _c<0>)...))>;
             type r     = type(init);
             for(index_int j = 0; j < n; j++)
             {

